### PR TITLE
Handle links in XML checksum generation and improve tests auto actions

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | The core may now contain some symbolic links (like `admin-api/bundles/apiplatform`) we cannot compute the md5 value of the symlink especially if it's a link to a folder so we introduce a new type of XML export dedicated for symbolic links The upgrade module will have to handle this correctly For now we simply use an md5 of the link value but it may not be enough and will have to be improved<br>Also add a few improvement in UI tests action workflows like extra inputs for better reusability
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Run `php tools/build/CreateRelease.php --version="9.0.0"` there should be no longer some warnings and in the checksum a new `<link />` tag should now be present
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/10158372361
| Fixed issue or discussion?     | ~
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Review by tleon — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Comment by Quetzacoalt91** [comment]
Adding the "Need autoupgrade PR" label as we likely need to handle this new tag when computing md5 hashes, i.e when comparing altered files.

**Review by boherm — APPROVED** [review]

File: .docker/docker_run_git.sh
Comment (matthieu-rolland) [inline .docker/docker_run_git.sh:104]: is this really needed ? it's supposed to be set to true by default

Comment (matthieu-rolland) [reply .docker/docker_run_git.sh:104]: ah or is it something specific to UI tests where it's set to false and must be set to true ?

**Review by nicosomb — APPROVED** [review]

File: tools/build/Library/ReleaseCreator.php
Comment (matthieu-rolland) [inline tools/build/Library/ReleaseCreator.php:905]: I didn't know this string function ! `str_repeat` 

This is rarely used it seems, we learn new stuff everyday :D

**Review by matthieu-rolland — COMMENTED** [review]
just one question about the dev mode activation, LGTM otherwise

File: .docker/docker_run_git.sh
Comment (jolelievre) [reply .docker/docker_run_git.sh:104]: Yeah it's to test the classic build, since the tests are based on the export of the CreateRelease script which automatically sets the dev mode to false this part is needed to reset it back to true (for debug purposes)

**Review by matthieu-rolland — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]
Ok for UI Tests

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
